### DISCUSSION TITLE
Invert Y Coordinate System

### DISCUSSION THREAD

**Discussion by santuari** [fabricjs/fabric.js#6985] [Q&A]
We would like to show an Autocad draw in fabricjs by maintaining the same User Coordinate System and without changing the coordinates of the points.
In Autocad, the User Coordinate System is bottom left, instead in FabricJS is top left as is shown in the following image:
![image](https://user-images.githubusercontent.com/6616963/113316496-b32f9f80-930e-11eb-97a8-852b959b3c9b.png)

If we invert all the objects (flipY), the top coordinates are changed:
https://codepen.io/asd0rino/pen/VwPvGOq?editors=1011

What did we do wrong?
Thanks,
Michele

**Comment by santuari** [comment]
I have updated the code ped so that it should be more clear.

Could you please help us? 

Thanks!
Michele

**Reply by asturur** [reply]
I would imagine the solution is to flip the viewport rather than the objects themselves.
As of now flipping viewport aren't supported for interactivity, but should work for display.

Let me try an example in a jsfiddle

**Reply by asturur** [reply]
https://jsfiddle.net/nbf6x071/1/

Look at this, the image seems flipped, you can still control it.
I bet many things do not work tho.
But if it is for display, you can work with something like that

**Reply by santuari** [reply]
Thank you very much for your answer @asturur
Probably, as you said, it is not fully supported because we are not able to make it work in our code: no errors but the graph is not shown. 
Do you have any idea on how to debug this?

**Reply by asturur** [reply]
did you see my fiddle? flipping is with negative scale but the  tou have to add padding. Find the correct padding is up to you. Do you have a design that is a tad bigger than 2 small rects far away in the canvas?
